### ISSUE TITLE
[FEATURE REQUEST] Calculate BPM and add to the metadata option

### ISSUE THREAD

**Issue by Ceynou** [namidaco/namida#398]
**Is your feature request related to a problem? Please describe.**
Would be nice for example to have a workout session listening to music with a certain BPM, or even build a playlist with the BPM in mind.

**Describe the solution you'd like**
A global and per song option to calculate the BPM and add it to the metadata.

**Describe alternatives you've considered**
I guess I should add in the metadata from another application, I've seen MusicBee mentioned in another issue.

**Additional context**
Someone came up with a solution for an accurate BPM calculation and used it for his rhythm game editor, here's the paper:
https://github.com/nathanstep55/bpm-offset-detector/blob/main/original-paper/report.pdf

**Apps**
[ArrowVortex](https://arrowvortex.ddrnl.com/) is the mentioned editor.

**Comment by Ceynou** [comment]
> you are referring to the same BPM used to show the waveform right?

I have no idea what BPM means other than beat per minute, the tempo.

> what's the point on adding that to the track metadata while the values makes sense only to the library/app that extracted them?

And where do they extract it from? not all song has this metadata.


It's just a random suggestion, you're free to close it of course

**Comment by MSOB7YY** [comment]
we refer to it as "wavefom" and yes it is extracted once and cached as a separate file, the extraction is done by calculating peaks from the actual pcm data

**Comment by MSOB7YY** [comment]
my bad i got it completely wrong, see here for newer issue with similar request #1141
